### PR TITLE
fix: [CDS-53070]: Improved Jenkins job search to be more user friendly by using BiLevelSelect

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
before

https://user-images.githubusercontent.com/96036463/223202435-fa0de644-4880-42bb-979c-2be9d5d34399.mov

After

https://user-images.githubusercontent.com/96036463/223202494-8dd0a38a-4c7d-4896-83e4-6a16a698e30c.mov

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by sahithikolichala — DISMISSED** [review]

**Review by sahithikolichala — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepInputSet.tsx [hunk 1]
```diff
@@ -184,20 +191,46 @@ function JenkinsStepInputSet(formContentProps: any): JSX.Element {
...
+    return {
+      label: jobName,
+      value: jobName,
+      submenuItems: [],
+      hasSubmenuItems: false
+    } as SelectWithBiLevelOption
   }
 
-  const jobParameters = get(formik.values, `${prefix}spec.jobParameter`)
+  const jobNameRenderer = memoize((item: SelectWithBiLevelOption, { handleClick }) => (
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepInputSet.tsx:221]: Use `ItemRendererWithMenuItem`

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 2]
```diff
@@ -149,6 +149,7 @@ queueStep:
   queuedByResourceKey: 'Queued by Resource Key:'
   noQueuedExecutions: No queued executions
 jenkinsStep:
+  fetchingJobs: Fetching Jobs...
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/strings/strings.en.yaml:152]: We can use `loadingFieldOptions`

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/70-pipeline/strings/strings.en.yaml (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/strings/strings.en.yaml]: addressed

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepInputSet.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepInputSet.tsx]: addressed

**Pull request merged**